src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1481,5 +1489,15 @@ public static void SetTestHook(string property, bool value)
                 fieldInfo.SetValue(null, value);
             }
         }
+
+        /// <summary>This member is used for internal test purposes.</summary>
+        public static void SetTestHook(string property, int value)
+        {
+            var fieldInfo = typeof(InternalTestHooks).GetField(property, BindingFlags.Static | BindingFlags.NonPublic);
+            if (fieldInfo != null)
+            {
+                fieldInfo.SetValue(null, value);
```

Comment:
Maybe we should just change the existing method to take an `object` value. Boxing operation cannot be avoided anyway because `SetValue` method takes an object value. So it's better to not duplicate the code.